### PR TITLE
Remove deprecated `setFee` and `setLease` in `Transaction`

### PR DESCRIPTION
**Pull request by ahangsu**
Remove remaining deprecated methods in `Transaction`.

### CODE REVIEW

File: src/main/java/com/algorand/algosdk/account/Account.java [hunk 1]
```diff
@@ -247,7 +247,7 @@ static public void setFeeByFeePerByte(Transaction tx, BigInteger suggestedFeePer
         if (fee.compareTo(Account.MIN_TX_FEE_UALGOS) < 0) {
             fee = Account.MIN_TX_FEE_UALGOS;
         }
-        tx.setFee(fee);
+        tx.fee = fee;
```
Comment (algochoi) [inline src/main/java/com/algorand/algosdk/account/Account.java:250]: nit: a little strange that `tx.fee` is initialized twice in line 243 and here

**Review by algochoi — COMMENTED** [review]
Nice work, do you know if `setNote` should also be deprecated in favor of transaction builder?

File: src/main/java/com/algorand/algosdk/account/Account.java (hunk 1 above)
Comment (ahangsu) [reply src/main/java/com/algorand/algosdk/account/Account.java:248]: I am unaware of the previous initialization, but this function do not return early, so I am fine with just removing it.

Comment (ahangsu) [reply src/main/java/com/algorand/algosdk/account/Account.java:250]: ok it is interesting now, seems fee has something to do with `estimateEncodedSize`, so I don't think we should remove it...

Comment (algochoi) [reply src/main/java/com/algorand/algosdk/account/Account.java:250]: got it, sounds reasonable!

**Comment by michaeldiamant** [comment]
For reader awareness:
* `setFee` - Deprecation added in https://github.com/algorand/java-algorand-sdk/pull/85/.
* `setLease` - Deprecated added in https://github.com/algorand/java-algorand-sdk/pull/83.

File: CHANGELOG.md [hunk 2]
```diff
@@ -6,6 +6,7 @@
   of `TxGroup.assignGroupID(Address address, Transaction ...txns)`.
 * Remove `Account.transactionWithSuggestedFeePerByte` in favor of `Account.setFeeByFeePerByte`.
 * Remove deprecated methods in `Transaction.java`, mark public `Transaction` constructor as hidden in favor of `com.algorand.algosdk.builder.transaction`.
+* Remove deprecated `Transaction.setFee` and `Transaction.setLease` methods
```
Comment (michaeldiamant) [inline CHANGELOG.md:9]:
```suggestion
* Remove deprecated `Transaction.setFee` and `Transaction.setLease` methods.
```

**Review by michaeldiamant — APPROVED** [review]

**Pull request merged**